projects/components/src/table/table.component.ts

```diff
@@ -169,6 +170,21 @@ export class TableComponent
     onClick: (row: StatefulTableRow) => this.toggleRowExpanded(row)
   };
 
+  private readonly selectableRowColumnConfig: TableColumnConfig = {
+    field: '$$state',
```

Comment:
- Why is this single selectable column needed?
- Is it problematic that we're using $$state for multiple columns? I thought we need field to be unique.
- The naming should more clearly delineate that selectablecolumn and checkbox column are both selection modes.

Reply:
I _believe_ this will work because we only have one $$state column at a time... i.e. we can't have expandable rows with a checkbox, but that should be fine.

Reply:
but we can have expandable rows with single select right? Both of those use $$state

Reply:
Hmm, which screen is that?

Anyway, I actually think this will work now that I'm remembering some more. Don't think the `field` needs to be unique after all. If I remember correctly, its just used to pluck the relevant data from the row dictionary and hand that to the cell renderer. In which case, this will work for both selection and expansion.

But I'll let @skjindal93 confirm. I'm sure he tested. :-)

Reply:
It needs to be unique. Otherwise, it throws "Duplicate column definition name provided"

I made expandable and multi select mutually exclusive. Because, there aren't use cases yet.

Reply:
@aaron-steinfeld did you see a use case for this in the current mocks?

Reply:
> Why is this single selectable column needed?

Since, I added a new property `selected` in `TableRowState` and I wanted to maintain the `selected` property even for single selection mode (_to make it look complete_)

Hence, added an invisible single selectable column to maintain state

Reply:
> Hmm, which screen is that?

Any expandable screen really, right? Since single select is the default. The mutual exclusivity is only between the multiselect mode.


> Since, I added a new property selected in TableRowState and I wanted to maintain the selected property even for single selection mode (to make it look complete)


It makes sense to maintain it, but I'm not following how the column helps that

Reply:
> It makes sense to maintain it, but I'm not following how the column helps that

I thought there was a 1-1 mapping between the field `'$$state'` and the property defined in the StatefulTableRow interface

```
export interface StatefulTableRow extends TableRow {
  $$state: TableRowState;
}
```

Turns out, I was wrong

Reply:
> Any expandable screen really, right? Since single select is the default. The mutual exclusivity is only between the multiselect mode.

The single select still happens on the expandable screen by clicking on rows (_if that's what you are referring to_)

Reply:
> The single select still happens on the expandable screen by clicking on rows (if that's what you are referring to)
Right, but those screens would have two $$state columns - anyway, given the comment above about it not needing to be 1-1, it sounds like we can remove it and this conversation is moot.